controllers/chat/getChannelDetail.js

```diff
@@ -19,8 +20,7 @@ const getChannelDetail = async (req, res) => {
   }
   let {last_fetch_date = null} = req.query;
   if (last_fetch_date) {
-    last_fetch_date = new Date(last_fetch_date);
-    last_fetch_date = last_fetch_date.toISOString();
+    last_fetch_date = moment.utc(last_fetch_date).toDate().toISOString();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from using JavaScript's built-in Date object to using the `moment` library for date manipulation is a good one. The `moment` library provides a more robust and reliable way to handle dates across different time zones. However, it's important to note that the `utc()` function will convert the date to UTC time. If the `last_fetch_date` is already in UTC, this won't be an issue, but if it's in another timezone, this could potentially alter the date. Ensure that the `last_fetch_date` is always provided in UTC to avoid any potential issues.

```diff
- last_fetch_date = moment.utc(last_fetch_date).toDate().toISOString();
+ last_fetch_date = moment(last_fetch_date).utc().format();
```
This change ensures that the date is parsed first before being converted to UTC. This can help prevent potential issues if the input date is not in UTC.


<!-- This is an auto-generated comment by OSS CodeRabbit -->